src/util-inl.h

```diff
@@ -513,8 +513,9 @@ SlicedArguments::SlicedArguments(
 template <typename T, size_t S>
 ArrayBufferViewContents<T, S>::ArrayBufferViewContents(
     v8::Local<v8::Value> value) {
-  CHECK(value->IsArrayBufferView());
-  Read(value.As<v8::ArrayBufferView>());
+  CHECK(value->IsArrayBufferView() || value->IsSharedArrayBuffer() ||
```

Comment:
Do we need this CHECK or could this be a DCHECK? It seems like we do the exact same check in `ReadValue()` again?

Reply:
This can be a `DCHECK` since we already test it inside `ReadValue`. If you think this is a required change, I can commit and re-run the tests.

Reply:
I would expect it to minimally improve the performance further and would include the suggestions therefore.

Reply:
I updated my pull request. Thank you for the review!